### PR TITLE
Merge all documentation into one location

### PR DESCRIPTION
**Pull request by alexcrichton**
This PR is aimed at making the this repository, and the
`rustwasm.github.io` domain, the one stop shop for all documentation for
Rust and WebAssembly. Currently we have a number of sources of documentation:

* https://rustwasm.github.io/book - a tutorial and guide to Rust and WebAssembly
* https://rustwasm.github.io/wasm-bindgen - a tutorial and guide
  specific to `wasm-bindgen`
* https://rustwasm.github.io/wasm-pack - a tutorial and guide
  specific to `wasm-pack`
* https://rustwasm.github.io/ - the Rust and WebAssembly blog

This commit reorganizes all of these properties to instead be deployed
in this repository in one location, hopefully making it easier to
discover more of them and also cross-link between them. The setup here
is somewhat similar to the Rust documentation itself, comprising of a
number of books.

When this site is built it will clone the book, wasm-bindgen, and
wasm-pack repositories. The latter two will be reset to their most
recent tag, matching the most recently published version on crates.io.
All books are then compiled with `mdbook` and placed into the output here.

The current set is then laid out as follows:

* https://rustwasm.github.io/ - landing page, button to install
  wasm-pack, links everywhere else
* https://rustwasm.github.io/blog.html - new home for the blog
* https://rustwasm.github.io/docs.html - dispatch point for documentation

I'm hoping that we can centralize here by using
https://rustwasm.github.io if anyone wants an entry point or an
easy-click installer and using `/docs.html` as the main location for
documentation.

Once this lands I'd like to update the deployments of the book,
wasm-bindgen, and wasm-pack. All those books will gain a banner saying
they're a "nightly preview" with links to the officially published
versions. This means that visiting https://rustwasm.github.io/book will
have a banner saying you should go to
https://rustwasm.github.io/docs/book instead basically. The landing page
at https://rustwasm.github.io/wasm-pack will probably stay for now (and
the installer page will definitely stay), but we likely won't link to it
much any more.

Finally for a deployment strategy I hope to turn this into a daily cron
job for this repository. That means we'll redeploy everything at least
once a day, picking up any changes naturally. If we want to manually
trigger a redeploy, however, we can simply reexecute the previous build
on Travis.

### CODE REVIEW

File: docs.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+* [The rustwasm book](docs/book)
+* [The wasm-bindgen book](docs/wasm-bindgen)
+* [The wasm-pack book](docs/wasm-pack)
```
Comment (alexcrichton) [inline docs.md]: This page is very bare bones and much smaller than it would otherwise want to be before merging

**Comment by alexcrichton** [comment]
Unfortunately it wasn't easy to get a preview up and running online due to all the absolute urls here (like css and such). The best way to preview this I think is to check it out locally and run `ci/build.sh`

**Comment by fitzgen** [comment]
Does it make sense to pull all the books together into a single `mdbook`?

**Comment by alexcrichton** [comment]
I'd ideally like to do that! I'm not aware of an mdbook option to do that though :(

**Comment by alexcrichton** [comment]
@fitzgen ok removed the `/blog.html` page and merged it back to the main page

**Pull request merged**